File: sdk-ts/src/adapters/langchain.ts
/**
 * Explicit LangChain.js entry point. No import-time dependency on
 * @langchain/core — duck-types on the same `.name`/`.func`/`.call`/`.invoke`
 * shapes Guard.wrapTools already handles generically. Prefer
 * `guard.wrapTools(tools)` directly; this exists so LangChain.js users can
 * find the integration by name.
 */
import type { Guard } from "../guard.ts";

export function wrapTools(guard: Guard, tools: unknown[]): unknown[] {
  return guard.wrapTools(tools);
}

File: sdk-python/agentguard/adapters/langchain.py
"""Explicit LangChain entry point. This has no import-time dependency on
langchain itself — it duck-types on the same `.name`/`.func`/`._run` shapes
Guard.wrap_tools already handles generically. Prefer `guard.wrap_tools(tools)`
directly; this module exists so LangChain users can find the integration by
name and see it documented against the library they're using.
"""
from __future__ import annotations

from typing import Any, List

from ..guard import Guard


def wrap_tools(guard: Guard, tools: List[Any]) -> List[Any]:
    """Wrap a list of LangChain `Tool`/`BaseTool`/`StructuredTool` objects so
    every invocation is policy-checked first. Equivalent to
    `guard.wrap_tools(tools)`.
    """
    return guard.wrap_tools(tools)
